core/src/com/unciv/ui/worldscreen/WorldScreen.kt

```diff
@@ -822,8 +823,15 @@ class WorldScreen(val gameInfo: GameInfo, val viewingCiv:CivilizationInfo) : Bas
 
             else ->
                 NextTurnAction("${Fonts.turn}{Next turn}", Color.WHITE) {
-                    game.settings.addCompletedTutorialTask("Pass a turn")
-                    nextTurn()
+                    if(game.settings.confirmNextTurn) {
+                        YesNoPopup("Confirm next turn", {
+                            game.settings.addCompletedTutorialTask("Pass a turn")
+                            nextTurn()
+                        }, UncivGame.Current.worldScreen).open()
+                    } else {
+                        game.settings.addCompletedTutorialTask("Pass a turn")
+                        nextTurn()
```

Comment:
You can move these 2 lines to a separate function (e.g. `nextTurnClicked`) and reuse them, however, that's a minor thing.

Reply:
I was thinking about that but I wasn't sure if its worth it for just 2 lines of code which are only used in this if-else

Reply:
And `game.settings` could be `UncivGame.Current.settings`... or was that the other way round? Or can a `this` be used somewhere? Maybe this looks tasty:?
```kotlin
                    val action = {
                        game.settings.addCompletedTutorialTask("Pass a turn")
                        nextTurn()
                    }
                    if (setting)
                        YesNoPopup("Confirm", action, this).open()
                    else action()
```

Reply:
the game.settings was already used before so I just kept it as is. Its like this in the entire file
But I'll add the action like this cause it looks much better than an additional function